File: src/main/java/entity/Menu.java
package entity;

public class Menu extends Produit {
    private Burger burger;
    private Complement boisson;
    private Complement frites;

    public Menu() {
        super();
        this.typeProduit = TypeProduit.MENU;
    }

    public Menu(int id, String nom, Burger burger, Complement boisson, Complement frites) {
        super(id, nom, 0, TypeProduit.MENU);
        this.burger = burger;
        this.boisson = boisson;
        this.frites = frites;
        calculerPrix();
    }

    public Burger getBurger() {
        return burger;
    }

    public void setBurger(Burger burger) {
        this.burger = burger;
        calculerPrix();
    }

    public Complement getBoisson() {
        return boisson;
    }

    public void setBoisson(Complement boisson) {
        this.boisson = boisson;
        calculerPrix();
    }

    public Complement getFrites() {
        return frites;
    }

    public void setFrites(Complement frites) {
        this.frites = frites;
        calculerPrix();
    }

    public void calculerPrix() {
        this.prix = 0;
        if (burger != null) this.prix += burger.getPrix();
        if (boisson != null) this.prix += boisson.getPrix();
        if (frites != null) this.prix += frites.getPrix();
    }

    @Override
    public String toString() {
        String burgerNom = (burger != null) ? burger.getNom() : "N/A";
        String boissonNom = (boisson != null) ? boisson.getNom() : "N/A";
        String fritesNom = (frites != null) ? frites.getNom() : "N/A";
        return id + " | " + nom + " | " + prix + " FCFA | " + burgerNom + " + " + boissonNom + " + " + fritesNom;
    }
}

File: src/Entity/Menu.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
class Menu extends Produit
{
    #[ORM\ManyToOne(targetEntity: Burger::class)]
    #[ORM\JoinColumn(name: 'burger_id', referencedColumnName: 'id', nullable: true)]
    private ?Burger $burger = null;

    #[ORM\ManyToOne(targetEntity: Complement::class)]
    #[ORM\JoinColumn(name: 'boisson_id', referencedColumnName: 'id', nullable: true)]
    private ?Complement $boisson = null;

    #[ORM\ManyToOne(targetEntity: Complement::class)]
    #[ORM\JoinColumn(name: 'frites_id', referencedColumnName: 'id', nullable: true)]
    private ?Complement $frites = null;

    public function calculerPrix(): float
    {
        $total = 0.0;
        
        if ($this->burger) {
            $total += (float) $this->burger->getPrix();
        }
        if ($this->boisson) {
            $total += (float) $this->boisson->getPrix();
        }
        if ($this->frites) {
            $total += (float) $this->frites->getPrix();
        }
        
        return $total;
    }

    public function getBurger(): ?Burger
    {
        return $this->burger;
    }

    public function setBurger(?Burger $burger): static
    {
        $this->burger = $burger;
        return $this;
    }

    public function getBoisson(): ?Complement
    {
        return $this->boisson;
    }

    public function setBoisson(?Complement $boisson): static
    {
        $this->boisson = $boisson;
        return $this;
    }

    public function getFrites(): ?Complement
    {
        return $this->frites;
    }

    public function setFrites(?Complement $frites): static
    {
        $this->frites = $frites;
        return $this;
    }

    public function getComposition(): string
    {
        $parts = [];
        if ($this->burger) {
            $parts[] = $this->burger->getNom();
        }
        if ($this->boisson) {
            $parts[] = $this->boisson->getNom();
        }
        if ($this->frites) {
            $parts[] = $this->frites->getNom();
        }
        return implode(' + ', $parts);
    }
}
